### PR TITLE
OSSM-3731: Add maistra api postsubmit jobs

### PR DESCRIPTION
**Pull request by leoieggli**

### CODE REVIEW

**Comment by leoieggli** [comment]
/pj-rehearse pull-ci-maistra-api-maistra-2.3-api-update-istio-2.3

File: ci-operator/config/maistra/api/maistra-api-maistra-2.3.yaml [hunk 1]
```diff
@@ -34,15 +34,14 @@ tests:
   commands: |
     ./maistra/run-test-infra-script.sh ./tools/automator.sh \
     -o maistra \
-    -f /creds-github/github-token \
+    -f /creds-github/token \
     -r istio \
     -c UPDATE_BRANCH=$AUTOMATOR_SHA bin/update_maistra_deps.sh \
```
Comment (jwendell) [inline ci-operator/config/maistra/api/maistra-api-maistra-2.3.yaml:39]: remove the UPDATE_BRANCH variable:

`-c bin/update_maistra_deps.sh`

also for the 2.4 job below.

**Review by jwendell — CHANGES_REQUESTED** [review]

**Comment by leoieggli** [comment]
/pj-rehearse pull-ci-maistra-api-maistra-2.3-api-update-istio-2.3

**Comment by leoieggli** [comment]
/pj-rehearse pull-ci-maistra-api-maistra-2.4-api-update-istio-2.4

**Pull request closed without merging**